touchscreen.lua

```diff
@@ -161,7 +163,22 @@ local function create_formspec(meta, data)
 	if focus then
 		fs = fs.."set_focus["..focus.."]"
 	end
-	return fs..table.concat(data)
+
+	local data_size = 0
+	for idx, val in pairs(data) do
+		if idx > data_size then
+			data_size = idx
+		end
+	end
+	
+	local formspec_result = fs
+	for i = 1, data_size, 1 do
+		if data[i] then
+			formspec_result = formspec_result..data[i]
+		end
+	end
+	
+	return formspec_result
```

Comment:
Less code, and without whitespace.
```suggestion
	local data_size = 0
	for i in pairs(data) do
		if i > data_size then
			data_size = i
		end
	end
	for i=1, data_size do
		if data[i] then
			fs = fs..data[i]
		end
	end
	return fs
```